core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -0,0 +1,529 @@
+    }
+
+    topicOverview.setTopicInfoList(topicInfoList);
+    return aclInfo;
+  }
+
+  private void enrichTopicInfoList(
+      String topicNameSearch,
+      HandleDbRequests handleDb,
+      int tenantId,
+      Integer loggedInUserTeam,
+      ArrayList<String> reqTopicsEnvsList,
+      List<TopicInfo> topicInfoList,
+      List<AclInfo> aclInfo,
+      List<AclInfo> prefixedAclsInfo,
+      Integer topicOwnerTeam) {
+    List<Acl> prefixedAcls = new ArrayList<>();
+    List<Acl> aclsFromSOT = new ArrayList<>();
+    List<Acl> allPrefixedAcls;
+    List<AclInfo> tmpAclPrefixed;
+    List<AclInfo> tmpAcl;
+    for (TopicInfo topicInfo : topicInfoList) {
+      aclsFromSOT.addAll(
+          getAclsFromSOT(topicInfo.getClusterId(), topicNameSearch, false, tenantId));
+
+      tmpAcl =
+          applyFiltersAclsForSOT(loggedInUserTeam, aclsFromSOT, tenantId).stream()
+              .collect(Collectors.groupingBy(AclInfo::getTopicname))
+              .get(topicNameSearch);
+
+      if (tmpAcl != null) {
+        aclInfo.addAll(tmpAcl);
+      }
+
+      allPrefixedAcls = handleDb.getPrefixedAclsSOT(topicInfo.getClusterId(), tenantId);
+      if (allPrefixedAcls != null && allPrefixedAcls.size() > 0) {
+        for (Acl allPrefixedAcl : allPrefixedAcls) {
+          if (topicNameSearch.startsWith(allPrefixedAcl.getTopicname())) {
+            prefixedAcls.add(allPrefixedAcl);
+          }
+        }
+        tmpAclPrefixed = applyFiltersAclsForSOT(loggedInUserTeam, prefixedAcls, tenantId);
+        prefixedAclsInfo.addAll(tmpAclPrefixed);
+      }
+
+      // show edit button only for restricted envs
+      if (Objects.equals(topicOwnerTeam, loggedInUserTeam)
+          && reqTopicsEnvsList.contains(topicInfo.getClusterId())) {
+        topicInfo.setShowEditTopic(true);
+      }
+    }
+  }
+
+  private void enrichTopicOverview(
+      int tenantId,
+      List<Topic> topics,
+      TopicOverview topicOverview,
+      String syncCluster,
+      List<TopicInfo> topicInfoList,
+      List<TopicHistory> topicHistoryList) {
+    ArrayList<TopicHistory> topicHistoryFromTopic;
+    for (Topic topic : topics) {
+      TopicInfo topicInfo = new TopicInfo();
+      topicInfo.setCluster(getEnvDetails(topic.getEnvironment(), tenantId).getName());
+      topicInfo.setClusterId(topic.getEnvironment());
+      topicInfo.setNoOfPartitions(topic.getNoOfPartitions());
+      topicInfo.setNoOfReplcias(topic.getNoOfReplcias());
+      topicInfo.setTeamname(manageDatabase.getTeamNameFromTeamId(tenantId, topic.getTeamId()));
+
+      if (syncCluster != null && syncCluster.equals(topic.getEnvironment())) {
+        topicOverview.setTopicDocumentation(topic.getDocumentation());
+        topicOverview.setTopicIdForDocumentation(topic.getTopicid());
+      }
+
+      if (topic.getHistory() != null) {
+        try {
+          topicHistoryFromTopic =
+              OBJECT_MAPPER.readValue(topic.getHistory(), new TypeReference<>() {});
```

Comment:
This could be extracted into constant

Reply:
There is only one occurrence of this. May be its ok to keep it as is.

Reply:
however it is inside a loop => a new object will be created at almost each iteration (if `if` conditions pass)
And also at each call of this method

Reply:
got it, updated.